### PR TITLE
OM-3504: Add `SearchBar` component

### PR DESCRIPTION
**Pull request by amar-hunar**
This PR addresses [OM-3504](https://hunar.atlassian.net/browse/OM-3504)

## Features:

- Add `SearchBar` component

## Links:
- [Design Link](https://www.figma.com/design/pRdpUhL25JXTmRVC1eF8t8/JQ-Features-%7C-Hand-Off?node-id=1770-37917&node-type=frame&m=dev)
- [Storybook Link](https://search-bar--666ac08efe69dd9c59a1e4c6.chromatic.com/?path=/docs/components-searchbar--documentation)

### CODE REVIEW

File: src/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+} from '@phosphor-icons/react';
+
+import { SearchBarField } from './SearchBarField';
+
+import { useIsMobile } from '@/hooks/useIsMobile';
+
+export interface SearchBarProps {
+    placeholder?: string;
+    showMobileSearch?: boolean;
+    mobileSearchWidth?: string;
```
Comment (misss-popcorn) [inline src/components/SearchBar/SearchBar.tsx:18]:
```suggestion
    xsWidth?: string;
```

Comment (misss-popcorn) [reply src/components/SearchBar/SearchBar.tsx:18]: what does showMobileSearch means, pls attach the relevant screenshot

Comment (amar-hunar) [reply src/components/SearchBar/SearchBar.tsx]: `showMobileSearch` controls whether mobile version of the search bar is shown on mobile or not. By default it is `true`. When it is `false`, Mobile version of the search bar is not shown on Mobile screen (refer image). This functionality is required for showing search bar appropriate on careers and referral page on mobile screen

image

Comment (amar-hunar) [reply src/components/SearchBar/SearchBar.tsx]: Please find the neccessary screenshots which shows the behaviour of `showMobileSearch`

## Mobile Screen:
- `showMobileSearch={true}`
image

- `showMobileSearch={false}`
image

## Desktop screen:
- `showMobileSearch={true}`
image

- `showMobileSearch={false}`
image

File: src/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    MagnifyingGlass as MagnifyingGlassIcon
+} from '@phosphor-icons/react';
+
+import { SearchBarField } from './SearchBarField';
+
+import { useIsMobile } from '@/hooks/useIsMobile';
+
+export interface SearchBarProps {
+    placeholder?: string;
+    showMobileSearch?: boolean;
```
Comment (misss-popcorn) [inline src/components/SearchBar/SearchBar.tsx:17]: why do we need this prop, should this search not always render mobile search on mobile?

Comment (amar-hunar) [reply src/components/SearchBar/SearchBar.tsx:17]: In Careers Page, Referral Page, we do not render the mobile search on mobile 👇 
image

**Review by misss-popcorn — APPROVED** [review]

**Pull request merged**